firmware/src/hal/sdl/link_sdl.cpp

```
@@ -0,0 +1,214 @@
+
+void send_byte(uint8_t byte) {
+  sendto(sock_, &byte, 1, 0, reinterpret_cast<sockaddr*>(&peer_), sizeof peer_);
+}
+
+// The wire carries one byte every byte time; a caller offering faster is refused and
+// tries again, so a clock pulse and a pattern share the 31250 baud a real cable has.
+bool wire_free(uint64_t now) { return now >= next_send_us_; }
+void wire_took(uint64_t now) { next_send_us_ = now + hal::kMidiByteUs; }
+
+int parse_ports(const char* spec, int& bind_port, int& peer_port) {
+  return std::sscanf(spec, "%d:%d", &bind_port, &peer_port) == 2 ? 0 : -1;
```

Clang (14.0.6)

[warning] do not call c-style vararg functions

(cppcoreguidelines-pro-type-vararg)